src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/SlayerFeatures.kt

```diff
@@ -1160,9 +1192,125 @@ object SlayerFeatures : CoroutineScope {
         var totemEntity: EntityArmorStand? = null
         var totemPos: BlockPos? = null
 
+        private var lastTickInvis = false
+        val relevantEntity: Entity?
+            get() {
+                return if (entity.isInvisible) {
+                    if (quazii == null || typhoeus == null) {
+                        null
+                    } else if (typhoeusTimer?.displayName?.formattedText?.contains("IMMUNE") == true
+                        || (typhoeus?.health ?: 0f) <= 0f) {
+                        quazii
+                    } else {
+                        typhoeus
+                    }
+                } else {
+                    entity
+                }
+            }
+        val relevantColor: Color?
+            get() {
+                val relevantTimer = if (entity.isInvisible) {
+                    if (quazii == null || typhoeus == null) {
+                        null
+                    } else if (typhoeusTimer?.displayName?.formattedText?.contains("IMMUNE") == true
+                        || (typhoeus?.health ?: 0f) <= 0f) {
+                        quaziiTimer
+                    } else {
+                        typhoeusTimer
+                    }
+                } else {
+                    timerEntity
+                } ?: return null
+                val attunement = relevantTimer.displayName.unformattedText.substringBefore(" ").stripControlCodes()
+                return attunementColors[attunement]
+            }
```

Comment:
How often are these accessed? May be worth handling it somewhere else.

Reply:
As often as it gets; I haven't had any problems of performance loss but could check for armor stands being renamed from the packet listener to update the relevant entity. Would rather make a few more tests with a profiler but I don't have enough time to make a proper extensive test atm

Reply:
After around half an hour of each boss being spawned, neither enderman nor blaze bosses made enough of an impact to appear in the profiler. I haven't noticed any increase on CPU or Memory usage or any FPS drop.
Although a single test isn't enough to decide on the impact of something like this, I think it's more worthwhile to manage all potential bottlenecks in a full rewrite of the file in the feature/slayer branch.